### ISSUE TITLE
High ram usage after playing a few matches

### ISSUE THREAD

**Issue by user1-github** [redeclipse/base#1580]
Game version - 2.0.9-9 Steam beta, Linux.

I noticed that after playing a few matches and then exiting the game, my ram usage is abnormally high - about 1.5 - 2.0 gb higher than before I launched the game. I don't see such increase in ram usage after playing older versions of Red Eclipse (1.6.0 or 2.0) or after using any other software on my system, so it seems like there might be a memory leak/s in the game.

How to reproduce:
- Enter the game (make sure it's a cold launch of the game)
- Play a few matches on different maps (you don't have to play till the end)
- Exit the game and check ram usage

**Comment by q009** [comment]
That sounds like OS shenanigans. If the Red Eclipse process is not running, it's literally up to the OS to figure out what to do with the memory that was used before.

**Comment by user1-github** [comment]
> Please check if the process is running when you exit the game.

I verified that the process isn't running after exiting the game.

About a year ago I tried one of the master Appimages of the game (when they were still being built). From what I remember, the game was already pretty close to its current state when I tried the master Appimage, but I also remember the same symptom of high ram usage after I exited the game, just like now.

If you don't experience this, I assume it might be due to differences in how different desktop environments manage memory. I use Gnome, so idk if I'll experience the same symptom on a different desktop environment. On the other hand, 1.5 - 2.0 gb higher ram usage than before seems too much, especially considering I don't experience this with any other software on my system, so I think this can't be blamed entirely on the desktop environment.

**Comment by user1-github** [comment]
> Specifically, how do you verify the amount of memory used and amount of memory being free?

I check it in Gnome system monitor.
For example, I remember before I played RE, my ram usage was 1.7 GB. After I played RE, I checked ram usage in Gnome system monitor again and now it's 4.2.

**Comment by q009** [comment]
Please check with ``free -h`` in your terminal. Don't look at the "free" value, look for "available".

**Comment by user1-github** [comment]
> Please check with `free -h` in your terminal. Don't look at the "free" value, look for "available".

Ok, so I tested the game again for ram usage.

Before playing RE, free -h showed 29gb under "available".

I played 5 matches on different maps in offline mode.

After exiting, free -h showed 27gb under "available".

Also checked Gnome monitor again just for comparison. Before playing it showed 1.4gb, after playing 4.5gb. So it's pretty consistent with what free -h shows.
